snapcraft/plugins/catkin.py

```diff
@@ -919,3 +961,58 @@ def _get_highest_version_path(path):
         raise RuntimeError('nothing found in {!r}'.format(path))
 
     return paths[-1]
+
+
+class _Wstool:
+    def __init__(self, ros_package_path, wstool_path, ubuntu_sources, project):
+        self._ros_package_path = ros_package_path
+        self._ubuntu_sources = ubuntu_sources
+        self._wstool_path = wstool_path
+        self._wstool_install_path = os.path.join(wstool_path, 'install')
+        self._project = project
+
+    def setup(self):
+        os.makedirs(self._wstool_install_path, exist_ok=True)
+
+        # wstool isn't a dependency of the project, so we'll unpack it
+        # somewhere else, and use it from there.
```

Comment:
If we want reproducible builds, we have to track them. But if they are not part of the build packages, then this has to be part of the future work to let plugins save stuff in the state that will then end up in prime/snap/snapcraft.yaml. Nothing to do for now I guess.